lib/rules/_base.js

```diff
@@ -521,16 +522,32 @@ module.exports = class {
       defaults.filePath = this.filePath;
     }
 
-    // perform the node prop expansion only if those properties don't exist in result
+    if (!result.node) {
+      if (!loggedRules.has(this.ruleName)) {
+        let message = `Calling the log method without passing the node is deprecated. Please ensure you pass node in the log method's result.`;
+
+        if (process.env.EMBER_TEMPLATE_LINT_DEV_MODE === '1') {
+          throw new Error(message);
+        } else {
+          this._console.log(message);
+
+          loggedRules.add(this.ruleName);
+        }
+      }
+    }
+
+    // perform the node property expansion only if those properties don't exist in result
     if (result.node) {
       let node = result.node;
+      delete result.node;
+      // if we've passed node, we want to ensure that we're correctly expanding the properties
+      // to include line, column, and source. If the rule has passed custom values for these,
+      // we want to respect those.
       nodeProps = {
-        line: node.loc && node.loc.start.line,
-        column: node.loc && node.loc.start.column,
-        source: this.sourceForNode(node),
+        line: 'line' in result || (node.loc && node.loc.start.line),
+        column: 'column' in result || (node.loc && node.loc.start.column),
+        source: 'source' in result || this.sourceForNode(node),
```

Comment:
```suggestion
        line: 'line' in result ? result.line : (node.loc && node.loc.start.line),
        column: 'column' in result ? result.column : (node.loc && node.loc.start.column),
        source: 'source' in result ? result.source : this.sourceForNode(node),
```

Reply:
Oh derp. What a dork I am. Thanks for catching that obvious mistake.